src/cli/runtime/mod.rs

```diff
@@ -33,6 +34,30 @@ fn init_hook_http(http: Arc<impl HttpIO>, script: Option<blueprint::Script>) ->
     http
 }
 
+fn init_http_worker_io(script: Option<blueprint::Script>) -> Arc<dyn WorkerIO<Event, Command>> {
+    #[cfg(feature = "js")]
+    if let Some(script) = script {
+        return Arc::new(super::javascript::Runtime::new(script));
+    };
+
+    #[cfg(not(feature = "js"))]
+    tracing::warn!("JS capabilities are disabled in this build");
+    Arc::new(crate::core::javascript::DefaultJsRuntime {})
+}
+
+fn init_resolver_worker_io(
+    script: Option<blueprint::Script>,
+) -> Arc<dyn WorkerIO<async_graphql::Value, async_graphql::Value>> {
+    #[cfg(feature = "js")]
+    if let Some(script) = script {
+        return Arc::new(super::javascript::Runtime::new(script));
+    };
+
+    #[cfg(not(feature = "js"))]
+    tracing::warn!("JS capabilities are disabled in this build");
```

Comment:
This will log twice. May be move this logic to where TargetRuntime is being initialized.